### DISCUSSION TITLE
Using 1Wire temp sensors with Halmet

### DISCUSSION THREAD

**Discussion by ESchlotfeldt** [hatlabs/discussions#127] [Q&A] [answered]
Hello,
My code compile bombs out when trying to add 1Wire temp sensors to HALMET-example-firmware-dev3. Error messages are below:

<img width="839" alt="image" src="https://github.com/hatlabs/discussions/assets/106944628/cf220871-3ec1-436e-b449-77c6318a6e5c">

To set this up I added:
- the following library to platform.ini: SensESP/OneWire @ ^2.0.0
- the following header file to main.cpp: #include "sensesp_onewire/onewire_temperature.h"
- proven code to read the temp sensors

The configuration stuff and code was pasted from a previous build using SensESP 2.7.2 , and was working well.

Would really appreciate feedback as to what the issue might be.

**Comment by mairas** [comment, marked as answer]
SensESP v3 gets rid of the `Startable` class that was used in previous versions. The OneWireTemperature class inherits from `Startable` and thus fails.

I've created a dev-3 branch for SensESP/OneWire as well. Change your OneWire dependency in `platformio.ini` to `https://github.com/SensESP/OneWire.git#dev-3` and you should be good to go.

I probably should re-add `Startable` as a dummy class because this might be a very common problem...

**Reply by ESchlotfeldt** [reply]
OK thanks very much. 

One more thing: when do you expect V3 to go live? I'm a bit worried that I'm going out on a bit of a limb with a development version

**Reply by mairas** [reply]
With the most recent logging changes, I think all features/changes are in. I still want to kick the tyres a bit by doing a couple of new sensors on my own boat and updating the old ones. That'll give me some confirmation that there are no gaping holes remaining. And some documentation needs to be written. I'd say a few weeks, depending on how much other stuff I have on my plate.

**Reply by ESchlotfeldt** [reply]
Great, thanks👍.
